src/components/map/MapComponent.vue

```diff
@@ -195,68 +201,102 @@ function clearLayerAndApplyStyle(
   }
 }
 
+async function updateTraveltimeArrow() {
+  // Arrow style for travel time
+  const arrowLayer = getScratchLayer(vcsApp, RENNES_LAYER._traveltimeArrow)
+  let lineStrings: LineString[] = []
+
+  if (
+    viewStore.currentView === viewList.traveltimes &&
+    travelTimesViewStore.selectedTravelTime
+  ) {
+    lineStrings = lineStringsFromTraveltimes(
+      [travelTimesViewStore.selectedTravelTime],
+      vcsApp
+    )
+  } else if (viewStore.currentView === viewList.line) {
+    if (lineViewStore.selectedTravelTime) {
+      lineStrings = lineStringsFromTraveltimes(
+        [lineViewStore.selectedTravelTime],
+        vcsApp
+      )
```

Comment:
If a common store is used for the selectedTravelTime variable, we can factorize the code here:

```
if(TheNameOfTheNewStore.selectedTravelTime){
   lineStrings = lineStringsFromTraveltimes(
      [travelTimesViewStore.selectedTravelTime],
      vcsApp
    )
}
else if(viewStore.currentView === viewList.line) {
      const travelTimes = await apiClientService.fetchTravelTimeByLine(
          lineViewStore.selectedLine
      )
      lineStrings = lineStringsFromTraveltimes(travelTimes, vcsApp)
}
```